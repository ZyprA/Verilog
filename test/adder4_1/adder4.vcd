$date
	Sun Feb  2 15:04:50 2025
$end
$version
	Icarus Verilog
$end
$timescale
	1ns
$end
$scope module adder4_sim $end
$var wire 4 ! s [3:0] $end
$var reg 4 " a [3:0] $end
$var reg 4 # b [3:0] $end
$scope module a4 $end
$var wire 4 $ a [3:0] $end
$var wire 4 % b [3:0] $end
$var wire 4 & s [3:0] $end
$var wire 3 ' c [2:0] $end
$upscope $end
$upscope $end
$enddefinitions $end
$comment Show the parameter values. $end
$dumpall
$end
#0
$dumpvars
b0 '
b0 &
b0 %
b0 $
b0 #
b0 "
b0 !
$end
#100
b1 !
b1 &
b1 "
b1 $
#200
b10 !
b10 &
b10 "
b10 $
#300
b110 '
b1001 !
b1001 &
b111 #
b111 %
#400
b100 !
b100 &
b111 '
b1101 "
b1101 $
#500
b10 !
b10 &
b1011 "
b1011 $
#600
