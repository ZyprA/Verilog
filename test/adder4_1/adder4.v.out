#! /opt/homebrew/Cellar/icarus-verilog/12.0/bin/vvp
:ivl_version "12.0 (stable)";
:ivl_delay_selection "TYPICAL";
:vpi_time_precision - 9;
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/system.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/vhdl_sys.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/vhdl_textio.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/v2005_math.vpi";
:vpi_module "/opt/homebrew/Cellar/icarus-verilog/12.0/lib/ivl/va_math.vpi";
S_0x14360ac50 .scope module, "adder4_sim" "adder4_sim" 2 18;
 .timescale -9 -9;
v0x14361f1f0_0 .var "a", 3 0;
v0x14361f280_0 .var "b", 3 0;
v0x14361f310_0 .net "s", 3 0, L_0x143621890;  1 drivers
S_0x14360adc0 .scope module, "a4" "adder4" 2 22, 2 1 0, S_0x14360ac50;
 .timescale 0 0;
    .port_info 0 /INPUT 4 "a";
    .port_info 1 /INPUT 4 "b";
    .port_info 2 /OUTPUT 4 "s";
L_0x14361f580 .functor XOR 1, L_0x14361f3a0, L_0x14361f4a0, C4<0>, C4<0>;
L_0x14361f7f0 .functor AND 1, L_0x14361f690, L_0x14361f730, C4<1>, C4<1>;
L_0x14361fb60 .functor XOR 1, L_0x14361f920, L_0x14361fa40, C4<0>, C4<0>;
L_0x14361fd30 .functor XOR 1, L_0x14361fb60, L_0x14361fc30, C4<0>, C4<0>;
L_0x14361fff0 .functor AND 1, L_0x14361fe60, L_0x14361ff00, C4<1>, C4<1>;
L_0x1436202b0 .functor AND 1, L_0x1436200d0, L_0x143620170, C4<1>, C4<1>;
L_0x143620360 .functor OR 1, L_0x14361fff0, L_0x1436202b0, C4<0>, C4<0>;
L_0x143620640 .functor AND 1, L_0x143620490, L_0x143620530, C4<1>, C4<1>;
L_0x143620710 .functor OR 1, L_0x143620360, L_0x143620640, C4<0>, C4<0>;
L_0x1436205d0 .functor XOR 1, L_0x143620850, L_0x1436208f0, C4<0>, C4<0>;
L_0x143620bb0 .functor XOR 1, L_0x1436205d0, L_0x143620a90, C4<0>, C4<0>;
L_0x14361fae0 .functor AND 1, L_0x143620e30, L_0x143620fd0, C4<1>, C4<1>;
L_0x143620da0 .functor AND 1, L_0x143621210, L_0x1436212b0, C4<1>, C4<1>;
L_0x143621470 .functor OR 1, L_0x14361fae0, L_0x143620da0, C4<0>, C4<0>;
L_0x143621720 .functor AND 1, L_0x143621520, L_0x1436215c0, C4<1>, C4<1>;
L_0x143621400 .functor OR 1, L_0x143621470, L_0x143621720, C4<0>, C4<0>;
L_0x143621660 .functor XOR 1, L_0x143621a30, L_0x143621ba0, C4<0>, C4<0>;
L_0x143620b30 .functor XOR 1, L_0x143621660, L_0x143621d10, C4<0>, C4<0>;
v0x14360d490_0 .net *"_ivl_100", 0 0, L_0x143620b30;  1 drivers
v0x14361cf90_0 .net *"_ivl_11", 0 0, L_0x14361f690;  1 drivers
v0x14361d040_0 .net *"_ivl_13", 0 0, L_0x14361f730;  1 drivers
v0x14361d100_0 .net *"_ivl_14", 0 0, L_0x14361f7f0;  1 drivers
v0x14361d1b0_0 .net *"_ivl_19", 0 0, L_0x14361f920;  1 drivers
v0x14361d2a0_0 .net *"_ivl_21", 0 0, L_0x14361fa40;  1 drivers
v0x14361d350_0 .net *"_ivl_22", 0 0, L_0x14361fb60;  1 drivers
v0x14361d400_0 .net *"_ivl_25", 0 0, L_0x14361fc30;  1 drivers
v0x14361d4b0_0 .net *"_ivl_26", 0 0, L_0x14361fd30;  1 drivers
v0x14361d5c0_0 .net *"_ivl_3", 0 0, L_0x14361f3a0;  1 drivers
v0x14361d670_0 .net *"_ivl_31", 0 0, L_0x14361fe60;  1 drivers
v0x14361d720_0 .net *"_ivl_33", 0 0, L_0x14361ff00;  1 drivers
v0x14361d7d0_0 .net *"_ivl_34", 0 0, L_0x14361fff0;  1 drivers
v0x14361d880_0 .net *"_ivl_37", 0 0, L_0x1436200d0;  1 drivers
v0x14361d930_0 .net *"_ivl_39", 0 0, L_0x143620170;  1 drivers
v0x14361d9e0_0 .net *"_ivl_40", 0 0, L_0x1436202b0;  1 drivers
v0x14361da90_0 .net *"_ivl_42", 0 0, L_0x143620360;  1 drivers
v0x14361dc20_0 .net *"_ivl_45", 0 0, L_0x143620490;  1 drivers
v0x14361dcb0_0 .net *"_ivl_47", 0 0, L_0x143620530;  1 drivers
v0x14361dd60_0 .net *"_ivl_48", 0 0, L_0x143620640;  1 drivers
v0x14361de10_0 .net *"_ivl_5", 0 0, L_0x14361f4a0;  1 drivers
v0x14361dec0_0 .net *"_ivl_50", 0 0, L_0x143620710;  1 drivers
v0x14361df70_0 .net *"_ivl_55", 0 0, L_0x143620850;  1 drivers
v0x14361e020_0 .net *"_ivl_57", 0 0, L_0x1436208f0;  1 drivers
v0x14361e0d0_0 .net *"_ivl_58", 0 0, L_0x1436205d0;  1 drivers
v0x14361e180_0 .net *"_ivl_6", 0 0, L_0x14361f580;  1 drivers
v0x14361e230_0 .net *"_ivl_61", 0 0, L_0x143620a90;  1 drivers
v0x14361e2e0_0 .net *"_ivl_62", 0 0, L_0x143620bb0;  1 drivers
v0x14361e390_0 .net *"_ivl_68", 0 0, L_0x143620e30;  1 drivers
v0x14361e440_0 .net *"_ivl_70", 0 0, L_0x143620fd0;  1 drivers
v0x14361e4f0_0 .net *"_ivl_71", 0 0, L_0x14361fae0;  1 drivers
v0x14361e5a0_0 .net *"_ivl_74", 0 0, L_0x143621210;  1 drivers
v0x14361e650_0 .net *"_ivl_76", 0 0, L_0x1436212b0;  1 drivers
v0x14361db40_0 .net *"_ivl_77", 0 0, L_0x143620da0;  1 drivers
v0x14361e8e0_0 .net *"_ivl_79", 0 0, L_0x143621470;  1 drivers
v0x14361e970_0 .net *"_ivl_82", 0 0, L_0x143621520;  1 drivers
v0x14361ea10_0 .net *"_ivl_84", 0 0, L_0x1436215c0;  1 drivers
v0x14361eac0_0 .net *"_ivl_85", 0 0, L_0x143621720;  1 drivers
v0x14361eb70_0 .net *"_ivl_87", 0 0, L_0x143621400;  1 drivers
v0x14361ec20_0 .net *"_ivl_93", 0 0, L_0x143621a30;  1 drivers
v0x14361ecd0_0 .net *"_ivl_95", 0 0, L_0x143621ba0;  1 drivers
v0x14361ed80_0 .net *"_ivl_96", 0 0, L_0x143621660;  1 drivers
v0x14361ee30_0 .net *"_ivl_99", 0 0, L_0x143621d10;  1 drivers
v0x14361eee0_0 .net "a", 3 0, v0x14361f1f0_0;  1 drivers
v0x14361ef90_0 .net "b", 3 0, v0x14361f280_0;  1 drivers
v0x14361f040_0 .net "c", 2 0, L_0x143620cc0;  1 drivers
v0x14361f0f0_0 .net "s", 3 0, L_0x143621890;  alias, 1 drivers
L_0x14361f3a0 .part v0x14361f1f0_0, 0, 1;
L_0x14361f4a0 .part v0x14361f280_0, 0, 1;
L_0x14361f690 .part v0x14361f1f0_0, 0, 1;
L_0x14361f730 .part v0x14361f280_0, 0, 1;
L_0x14361f920 .part v0x14361f1f0_0, 1, 1;
L_0x14361fa40 .part v0x14361f280_0, 1, 1;
L_0x14361fc30 .part L_0x143620cc0, 0, 1;
L_0x14361fe60 .part v0x14361f1f0_0, 1, 1;
L_0x14361ff00 .part v0x14361f280_0, 1, 1;
L_0x1436200d0 .part v0x14361f280_0, 1, 1;
L_0x143620170 .part L_0x143620cc0, 0, 1;
L_0x143620490 .part L_0x143620cc0, 0, 1;
L_0x143620530 .part v0x14361f1f0_0, 1, 1;
L_0x143620850 .part v0x14361f1f0_0, 2, 1;
L_0x1436208f0 .part v0x14361f280_0, 2, 1;
L_0x143620a90 .part L_0x143620cc0, 1, 1;
L_0x143620cc0 .concat8 [ 1 1 1 0], L_0x14361f7f0, L_0x143620710, L_0x143621400;
L_0x143620e30 .part v0x14361f1f0_0, 2, 1;
L_0x143620fd0 .part v0x14361f280_0, 2, 1;
L_0x143621210 .part v0x14361f280_0, 2, 1;
L_0x1436212b0 .part L_0x143620cc0, 1, 1;
L_0x143621520 .part L_0x143620cc0, 1, 1;
L_0x1436215c0 .part v0x14361f1f0_0, 2, 1;
L_0x143621890 .concat8 [ 1 1 1 1], L_0x14361f580, L_0x14361fd30, L_0x143620bb0, L_0x143620b30;
L_0x143621a30 .part v0x14361f1f0_0, 3, 1;
L_0x143621ba0 .part v0x14361f280_0, 3, 1;
L_0x143621d10 .part L_0x143620cc0, 2, 1;
    .scope S_0x14360ac50;
T_0 ;
    %vpi_call 2 25 "$dumpfile", "adder4.vcd" {0 0 0};
    %vpi_call 2 26 "$dumpvars", 32'sb00000000000000000000000000000000, S_0x14360ac50 {0 0 0};
    %end;
    .thread T_0;
    .scope S_0x14360ac50;
T_1 ;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0x14361f1f0_0, 0, 4;
    %pushi/vec4 0, 0, 4;
    %store/vec4 v0x14361f280_0, 0, 4;
    %delay 100, 0;
    %pushi/vec4 1, 0, 4;
    %store/vec4 v0x14361f1f0_0, 0, 4;
    %delay 100, 0;
    %pushi/vec4 2, 0, 4;
    %store/vec4 v0x14361f1f0_0, 0, 4;
    %delay 100, 0;
    %pushi/vec4 7, 0, 4;
    %store/vec4 v0x14361f280_0, 0, 4;
    %delay 100, 0;
    %pushi/vec4 13, 0, 4;
    %store/vec4 v0x14361f1f0_0, 0, 4;
    %delay 100, 0;
    %pushi/vec4 11, 0, 4;
    %store/vec4 v0x14361f1f0_0, 0, 4;
    %delay 100, 0;
    %vpi_call 2 36 "$finish" {0 0 0};
    %end;
    .thread T_1;
# The file index is used to find the file name in the following table.
:file_names 3;
    "N/A";
    "<interactive>";
    "adder4.v";
